src/Mod/Draft/Draft.py

```diff
@@ -252,79 +252,14 @@ def makeRectangle(length, height, placement=None, face=None, support=None):
-    the dimension line passign through p3.The current line width and color
-    will be used. There are multiple  ways to create a dimension, depending on
-    the arguments you pass to it:
-    - (p1,p2,p3): creates a standard dimension from p1 to p2
-    - (object,i1,i2,p3): creates a linked dimension to the given object,
-    measuring the distance between its vertices indexed i1 and i2
-    - (object,i1,mode,p3): creates a linked dimension
-    to the given object, i1 is the index of the (curved) edge to measure,
-    and mode is either "radius" or "diameter".
-    """
-    if not FreeCAD.ActiveDocument:
-        FreeCAD.Console.PrintError("No active document. Aborting\n")
-        return
-    obj = FreeCAD.ActiveDocument.addObject("App::FeaturePython","Dimension")
-    _Dimension(obj)
-    if gui:
-        _ViewProviderDimension(obj.ViewObject)
-    if isinstance(p1,Vector) and isinstance(p2,Vector):
-        obj.Start = p1
-        obj.End = p2
-        if not p3:
-            p3 = p2.sub(p1)
-            p3.multiply(0.5)
-            p3 = p1.add(p3)
-    elif isinstance(p2,int) and isinstance(p3,int):
-        l = []
-        idx = (p2,p3)
-        l.append((p1,"Vertex"+str(p2+1)))
-        l.append((p1,"Vertex"+str(p3+1)))
-        obj.LinkedGeometry = l
-        obj.Support = p1
-        p3 = p4
-        if not p3:
-            v1 = obj.Base.Shape.Vertexes[idx[0]].Point
-            v2 = obj.Base.Shape.Vertexes[idx[1]].Point
-            p3 = v2.sub(v1)
-            p3.multiply(0.5)
-            p3 = v1.add(p3)
-    elif isinstance(p3,str):
-        l = []
-        l.append((p1,"Edge"+str(p2+1)))
-        if p3 == "radius":
-            #l.append((p1,"Center"))
-            if FreeCAD.GuiUp:
-                obj.ViewObject.Override = "R $dim"
-            obj.Diameter = False
-        elif p3 == "diameter":
-            #l.append((p1,"Diameter"))
-            if FreeCAD.GuiUp:
-                obj.ViewObject.Override = "Ø $dim"
-            obj.Diameter = True
-        obj.LinkedGeometry = l
-        obj.Support = p1
-        p3 = p4
-        if not p3:
-            p3 = p1.Shape.Edges[p2].Curve.Center.add(Vector(1,0,0))
-    obj.Dimline = p3
-    if hasattr(FreeCAD,"DraftWorkingPlane"):
-        normal = FreeCAD.DraftWorkingPlane.axis
-    else:
-        normal = FreeCAD.Vector(0,0,1)
-    if gui:
-        # invert the normal if we are viewing it from the back
-        vnorm = get3DView().getViewDirection()
-        if vnorm.getAngle(normal) < math.pi/2:
-            normal = normal.negative()
-    obj.Normal = normal
-    if gui:
-        formatObject(obj)
-        select(obj)
+from draftobjects.dimension import make_dimension
+makeDimension = make_dimension
+
+from draftobjects.dimension import LinearDimension
+_Dimension = LinearDimension
+
+from draftviewproviders.view_dimension import ViewProviderLinearDimension
+_ViewProviderDimension = ViewProviderLinearDimension
 
     return obj
```

Comment:
@carlopav You forgot to remove this `return obj` statement, and when the entire `Draft.py` is imported, it causes a syntax error. The Travis tests show this and fail because of it.